src/perftest_parameters.c

```diff
@@ -2641,26 +2635,54 @@ int parser(struct perftest_parameters *user_param,char *argv[], int argc)
 					CHECK_VALUE_NON_NEGATIVE(user_param->latency_gap,int,"Latency gap time",not_int_ptr);
 					latency_gap_flag = 0;
 				}
-#ifdef HAVE_CUDA
+				if (((use_cuda_flag || use_cuda_bus_id_flag) && !cuda_memory_supported()) ||
+				    (use_rocm_flag && !rocm_memory_supported()) ||
+				    (use_neuron_flag && !neuron_memory_supported()) ||
+				    (use_hl_flag && !hl_memory_supported())) {
+					printf(" Unsupported memory type\n");
+					return FAILURE;
+				}
+				if (user_param->memory_type != MEMORY_HOST &&
```

Comment:
Can you please add description for this if condition?

Reply:
Done.